### PR TITLE
[CI] Add noLD R test

### PR DESCRIPTION
**Pull request by hcho3**
Address comment https://github.com/dmlc/xgboost/pull/6378#issuecomment-725861901 https://github.com/dmlc/xgboost/pull/6378#issuecomment-725867810

* Add a unit test to cover the edge case that previously had failed in noLD R environment
* Use the Docker container `rhub/debian-gcc-devel-nold` to set up noLD R environment
* Make the test optional. The test must be manually invoked by adding a pull request comment `/gha run r-nold-test`

Example run: https://github.com/hcho3/xgboost/runs/1389096410

cc @jameslamb @vnijs

### CODE REVIEW

File: .github/workflows/r_nold.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+on:
+  pull_request_review_comment:
+    types: [created]
+
+env:
+  R_PACKAGES: c('XML', 'igraph', 'data.table', 'magrittr', 'ggplot2', 'DiagrammeR', 'Ckmeans.1d.dp', 'vcd', 'testthat', 'lintr', 'knitr', 'rmarkdown', 'e1071', 'cplm', 'devtools', 'float', 'titanic')
+
+jobs:
+  test-R-noLD:
+    if: github.event.comment.body == '/gha run r-nold-test' && contains('OWNER,MEMBER,COLLABORATOR', github.event.comment.author_association)
```
Comment (hcho3) [inline .github/workflows/r_nold.yml:15]: /gha run r-nold-test

Comment (hcho3) [reply .github/workflows/r_nold.yml:15]: ^ This comment triggers the job https://github.com/dmlc/xgboost/runs/1389351003

Comment (trivialfis) [reply .github/workflows/r_nold.yml:15]: You might want to document it.

Comment (hcho3) [reply .github/workflows/r_nold.yml:15]: Done.

File: .github/workflows/r_nold.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+on:
+  pull_request_review_comment:
+    types: [created]
+
+env:
+  R_PACKAGES: c('XML', 'igraph', 'data.table', 'magrittr', 'ggplot2', 'DiagrammeR', 'Ckmeans.1d.dp', 'vcd', 'testthat', 'lintr', 'knitr', 'rmarkdown', 'e1071', 'cplm', 'devtools', 'float', 'titanic')
+
+jobs:
+  test-R-noLD:
```
Comment (hcho3) [inline .github/workflows/r_nold.yml:14]: /gha run r-nold-test

File: doc/contrib/ci.rst [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+`.github/workflows <https://github.com/dmlc/xgboost/tree/master/.github/workflows>`_.
+
+Most of the tests listed in the configuration files run automatically for every incoming pull
+requests and every update to branches. A few tests however require manual activation:
+
+* R tests with ``noLD`` option: Run R tests using a custom-built R with compilation flag
+  ``--disable-long-double``. See `this page <https://blog.r-hub.io/2019/05/21/nold/>`_ for more
+  details about noLD. This is a requirement for keeping XGBoost on CRAN (the R package index).
+  To invoke this test suite for a particular pull request, simply add a review comment
+  ``/gha run r-nold-test``. (Ordinary comment won't work. It needs to be a review comment.)
```
Comment (hcho3) [inline doc/contrib/ci.rst:27]: /gha run r-nold-test

Comment (hcho3) [reply doc/contrib/ci.rst:27]: Note that the trigger needs to be a review comment, not an ordinary comment.

**Review by trivialfis — APPROVED** [review]

File: doc/contrib/ci.rst (hunk 3 above)
Comment (jameslamb) [reply doc/contrib/ci.rst:27]: I was confused about "review comment" at first when I set one of these up. I learned any comment on the "Files Changed" tab will work, even if you click "add single comment" and it isnt technically a review.

Replying to a comment thread started by such a comment will also trigger it.

File: .github/workflows/r_nold.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    - name: Install dependencies
+      shell: bash
+      run: |
+        cat > install_libs.R <<EOT
+          install.packages(${{ env.R_PACKAGES }},
+                           repos = 'http://cloud.r-project.org',
+                           dependencies = c('Depends', 'Imports', 'LinkingTo'))
+        EOT
+        /tmp/R-devel/bin/Rscript install_libs.R
```
Comment (jameslamb) [inline .github/workflows/r_nold.yml:37]: @hcho3 did you know you can make `shell` any executable? I think instead of dealing with `cat`-ing a file, this could be

```yaml
- name: Install dependencies
- shell: /tmp/R-devel/bin/Rscript {0}
- run: |
    install.packages(${{ env.R_PACKAGES }}, repos = "http://cloud.r-project.org", dependencies = c("Depends", "Imports", "LinkingTo"))
```

It might be a little simpler, if you want.

Comment (jameslamb) [reply .github/workflows/r_nold.yml:37]: I learned about that in `r-lib/actions` examples: https://github.com/r-lib/actions/blob/9598b8eeb6d88de7d76d580d84443542bbfdffce/examples/pkgdown.yaml#L26

Comment (hcho3) [reply .github/workflows/r_nold.yml:37]: @jameslamb `/tmp/R-devel/bin/Rscript {0}` resulted into a [strange error](https://github.com/hcho3/xgboost/runs/1389003461?check_suite_focus=true):
```
Error: Second path fragment must not be a drive or UNC name. (Parameter 'expression')
```

Comment (jameslamb) [reply .github/workflows/r_nold.yml:37]: oh sad! Maybe you can't use absolute paths there. Don't worry about it then, not that important either way.

**Comment by hcho3** [comment]
@trivialfis @jameslamb Thanks for reviewing!
